qualang_tools/results/results.py

```diff
@@ -68,11 +58,37 @@ def fetch_all(self):
             for data in self.data_list:
                 if hasattr(self.res_handles, data):
                     self.results.append(
-                        self.format_(self.res_handles.get(data).fetch_all())
+                        self._format(self.res_handles.get(data).fetch_all())
                     )
 
         elif self.mode == "live":
             self.results = []
             for i in range(len(self.data_handles)):
-                self.results.append(self.format_(self.data_handles[i].fetch_all()))
+                self.results.append(self._format(self.data_handles[i].fetch_all()))
         return self.results
+
+
+def progress_counter(iteration, total, progress_bar=True, percent=True, start=None):
+    """Displays progress bar and prints remaining computation time.
+
+    :param iteration: current iteration. Must be a python integer.
+    :param total: total number of iteration. Must be a python integer.
+    :param progress_bar: Flag enabling the progress bar display. Must be True or False. Default is True.
+    :param percent: Flag enabling the progress percentage display. Must be True or False. Default is True.
+    :param start: Starting time of the program enabling the elapsed time display. Must be the result of time.time(). Default is None.
+    :return: None.
+    """
+    current_percent = (iteration + 1) / total * 100
+
+    progress = "Progress: "
+    if progress_bar:
+        bar = "#" * int(current_percent / 2)
+        progress += "[" + bar + " " * (50 - len(bar)) + "] "
+    if percent:
+        progress += f"{current_percent:.1f}%"
+    if start is not None:
+        progress += f"--> elapsed time: {time.time()-start:.2f}s"
+
+    print(progress, end="\r")
+    if current_percent == 100:

```

Comment:
This will work poorly if someone stops a run in the middle...
Two things:
1. At the beginning have `print(end="\r")` and at the end (here) have `print(progress, end="")`. This would at least make it end at the end of the line, WDYT?
2. Getting a new line when the use interupts is hard... Maybe we need to add an empty print to the "on_close"?

Reply:
I tested it and it seems that the KeyboardInterrupt  call automatically goes to a new line in the console so I don't think there will be an issue here.
I tested your solution and the problem is that when the loop ends correctly, you do not start on a new line on the console which is what I wanted...

Reply:
If you tested it and it seems to work, then great :)